cmd/root.go

```diff
@@ -87,6 +90,7 @@ Examples:
 		PreRun: func(cmd *cobra.Command, args []string) {
 			viper.BindPFlags(cmd.Flags())
 		},
+		Version: version,
```

Comment:
Would like to read the version from [`debug.BuildInfo`](https://pkg.go.dev/runtime/debug#ReadBuildInfo) instead, so that regular `go install` also sets the version, and so I dont have to keep updating files containing the version.

Reply:
I have never used this feature before. I couldn't achieve reading the version from BuildInfo in my setup.

```go
func version() string {
	b, ok := debug.ReadBuildInfo()
	if !ok {
		return ""
	}
	return b.Main.Version
}
```

It returns `devel` all the time even when I set a new version tag in VCS.

Reply:
It's always "devel" when running locally. It's only set when using `go install` on a remote version. That's why I think goreleaser has this proxy setting that basically downloads the repo from the Go proxy just so the version gets set.

I suggest just using this method, and set the proxy flag in goreleaser.

If you want a more elaborate version when running locally, you could maybe look at the VCS fields from that struct and reconstruct a new version string, but only when the main version is `devel`. Such as `v0.4.0-devel+somegithash-20231001-dirty`

Reply:
I got it. It makes sense but I have a question. What would happen when the binary is delivered by 3. party package managers (besides `go install` I mean). I think we are assuming the binary is installed by `krew`. So we don't have to worry about the case when the `version` isn't set.

Reply:
Hmm the debug.BuildInfo isn't as "catch-all" of a solution as I'd hoped.

I still want regular `go install` to work (when providing remote path).

What about a hybrid solution? Use the `version` variable from main if it's set, otherwise fall back to debug.BuildInfo.

Seems like goreleaser automatically sets `main.version`[^1], so could probably skip that one from the `.goreleaser.yml` file. Less places needed to have the version written down.

[^1]: https://goreleaser.com/cookbooks/using-main.version/

Reply:
I pushed some changes, I hope you dont mind.

I had a new idea: what about relying on `go:embed`? If we're going to have the version written down somewhere (either in `.goreleaser.yml` or in the `Makefile`), then we might as well have a dedicated file for it.

The implementation still allows for `-X main.version=...`